### ISSUE TITLE
Update dictionary JSON spec

### ISSUE THREAD

**Issue by xicri** [xicri/genshin-langdata#40]
```json5
{
    slug: "inazuman", // auto generate
    en: {
       translation: "Inazuman",
       kana: "イナズマン",
       variants: [ "...", "...", ... ],
       note: "Originally, Inazuman was an unofficial...",
    },
    ja: {
       translation: [ "稲妻人", "稲妻の" ],
       kana: [ "いなずまじん", "いなずまの" ],
       variants: [ "...", "...", ... ],
       note: "元々 Inazuman は非公式に…(以下略)",
    },
    zhCN: {
       translation: [ "稻妻人", "稻妻的" ],
       // kana: [ "" ], 
       variants: [ "...", "...", ... ],
       note: "...",
    },
    
    "examples": [{
      "en": {
        sentence: "Inazumans are definitely more particular about etiquette than Mondstadters!",
        ref: "English title for the video",
        refURL: "https://youtube.com//...",
      },
      "ja": {
        sentence: "モンド人よりも、稲妻人の方が礼儀に対して気を配っている。",
        ref: "トーマ, キャラクター実戦紹介 トーマ「烈炎の守護」",
        refURL: "https://www.youtube.com/watch?v=jvmz4TrPgUE&t=16s",
      },
    }],

    "createdOn": "2022-01-01", // auto generate; At → On
    "updatedOn": "2022-01-01", // auto generate; At → On
    "tags": [ "inazuma" ]
  },
```

**Comment by Bill-Haku** [comment]
Hello, I am wondering when can this update be done. I have noticed the plan of adding pinyin the issue #114, and I also think it would be a lot helpful for users to add notes in English and Chinese, as they are mostly written in Japanese now.

**Comment by Bill-Haku** [comment]
Thanks for your work. BTW you can use a script to convert the existed json format to the new json format right?

**Comment by xicri** [comment]
@Bill-Haku

> BTW you can use a script to convert the existed json format to the new json format right?

Yes, I will write a small script to convert the existing JSON to the new format.
There is already the WIP repository: https://github.com/xicri/langdata-converter-v1

**Comment by xicri** [comment]
@Bill-Haku @SleepyAsh0191 I'm starting to update the JSON5 format of genshin-langdata to refactor and add new features such as English notes and `related` words.
If you have any input, let me know.
(I might make some minor updates for the new format later.)

Also, I'm considering conversion from JSON5 to TypeScript as I noted in the description of this issue. Can you let me know if you agree or disagree with conversion to TypeScript?

**Comment by Bill-Haku** [comment]
The new json format looks good, but please make a new API URI for the new format and keep the old format for a while so API users including me can update it and the users of the existed version will not be affected. 

Also, I think using TypeScript is a good choice. I have no problem with this conversion.

**Comment by xicri** [comment]
@Bill-Haku

> The new json format looks good,
> Also, I think using TypeScript is a good choice. I have no problem with this conversion.

Thanks!

> but please make a new API URI for the new format and keep the old format for a while so API users including me can update it and the users of the existed version will not be affected.

Yes, of course. I might update the URL for the old format (maybe something like https://dataset.genshin-dictionary.com/v0/words.json), but even if I update the URL, I will keep the old URL for one or some months after I release the new format to avoid sudden breaking changes for the developers including you.

**Comment by Bill-Haku** [comment]
Thank you. Looking forward to the update.
